app/Http/Controllers/Frontend/LeaderboardController.php

```diff
@@ -2,26 +2,55 @@
 
 namespace App\Http\Controllers\Frontend;
 
+use App\Enum\CacheKey;
 use App\Http\Controllers\Backend\LeaderboardController as LeaderboardBackend;
 use App\Http\Controllers\Controller;
 use Carbon\Carbon;
 use Illuminate\Contracts\Support\Renderable;
+use Illuminate\Support\Facades\Cache;
 
 class LeaderboardController extends Controller
 {
+    private static string $cacheRetentionConfigKey = 'trwl.cache.leaderboard-retention-seconds';
+
     public static function renderMonthlyLeaderboard(string $date): Renderable {
         $date = Carbon::parse($date);
+
+        $leaderboard = Cache::remember(
+            CacheKey::LeaderboardMonth . '-for-' . $date->toISOString(),
+            config(self::$cacheRetentionConfigKey),
+            fn() => LeaderboardBackend::getMonthlyLeaderboard($date));
+
         return view('leaderboard.month', [
-            'leaderboard' => LeaderboardBackend::getMonthlyLeaderboard($date),
+            'leaderboard' => $leaderboard,
             'date'        => Carbon::parse($date)
         ]);
     }
 
     public function renderLeaderboard(): Renderable {
+        $ttl = config(self::$cacheRetentionConfigKey);
+
+        $usersLeaderboard = Cache::remember(
+            CacheKey::LeaderboardGlobalPoints,
+            $ttl,
+            fn() => LeaderboardBackend::getLeaderboard());
+
+        $distanceLeaderboard = Cache::remember(
+            CacheKey::LeaderboardGlobalDistance,
+            $ttl,
+            fn() => LeaderboardBackend::getLeaderboard(orderBy: 'distance'));
+
+        $friendsLeaderboard = auth()->check()
+            ? Cache::remember(
+                CacheKey::getFriendsLeaderboardKey(auth()->id()),
+                $ttl,
+                fn() => LeaderboardBackend::getLeaderboard(onlyFollowings: true))
+            : null;
```

Comment:
This is hard to read.

```suggestion
        if(auth()->check()) {
                $friendsLeaderboard = Cache::remember(
                        CacheKey::getFriendsLeaderboardKey(auth()->id()),
                        $ttl,
                        fn() => LeaderboardBackend::getLeaderboard(onlyFollowings: true)
                );
        }
```

Reply:
Das finde ich Quatsch hingegen, weil wir dann konditional Variablen assignen und irgendwo später schauen müssen, ob die noch da sind. Wie kann sich denn jemand in 3 Jahren noch daran erinnern, warum man unten drunter nen `?? null` machen muss? Dann lieber hier oben direkt.

Reply:
Hmm. Wie du magst. Ich finde es so nur besser lesbar. nbfm